### PR TITLE
add omniauth-discord

### PR DESCRIPTION
**Pull request by yournumberone**

### CODE REVIEW

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 1]
```diff
@@ -29,4 +35,15 @@ def from_omniauth(auth)
       u.save
     end
   end
+
+  def from_omniauth_discord(auth)
```
Comment (Vadser) [inline app/controllers/users/omniauth_callbacks_controller.rb:39]: change the name of from_omniauth and remove some duplications from from_omniauth_* methods

**Review by Vadser — APPROVED** [review]

File: app/controllers/users/omniauth_callbacks_controller.rb (hunk 1 above)
Comment (yournumberone) [reply app/controllers/users/omniauth_callbacks_controller.rb:39]: done

**Comment by yournumberone** [comment]
upd

**Review by Vadser — DISMISSED** [review]

File: Gemfile [hunk 2]
```diff
@@ -37,18 +37,19 @@ end
...
+gem 'doorkeeper', '~> 5.6.0.rc2'
+gem 'doorkeeper-openid_connect', '~> 1.8', '>= 1.8.2'
 gem 'devise'
 gem 'omniauth', '~> 1.9'
+gem 'omniauth-discord'
 gem 'omniauth-yandex', github: 'evrone/omniauth-yandex', branch: 'master'
 gem 'pry'
 
 gem 'dotenv-rails'
+gem 'byebug'
```
Comment (anatoliliotych) [inline Gemfile:49]: please, move gems to dev/test group

File: config/initializers/doorkeeper.rb [hunk 3]
```diff
@@ -25,7 +25,7 @@
     end
     user
   end
-
+  # TODO: clear session!
```
Comment (anatoliliotych) [inline config/initializers/doorkeeper.rb:28]: ?

File: config/routes.rb [hunk 4]
```diff
@@ -26,6 +26,6 @@
   as :user do
     delete '/sign_out' => 'devise/sessions#destroy', :as => :destroy_user_session
   end
-  get '/login' => 'application#login', as: :login
+  get '/login' => 'application#login', as: :new_session
```
Comment (anatoliliotych) [inline config/routes.rb:29]: this change will probably require changes in other places like in alfred omniauth strategy, did you check this?

**Review by anatoliliotych — CHANGES_REQUESTED** [review]

File: config/routes.rb (hunk 4 above)
Comment (yournumberone) [reply config/routes.rb:29]: yes I did

File: config/initializers/doorkeeper.rb (hunk 3 above)
Comment (yournumberone) [reply config/initializers/doorkeeper.rb]: fixed

File: Gemfile (hunk 2 above)
Comment (yournumberone) [reply Gemfile]: fixed

**Pull request merged**